burnman/layer.py

```diff
@@ -0,0 +1,797 @@
+from __future__ import print_function
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2017 by the BurnMan team, released under the GNU
+# GPL v2 or later.
+
+import numpy as np
+from scipy.integrate import odeint
+from scipy.integrate import quad
+from scipy.interpolate import UnivariateSpline
+from burnman import averaging_schemes
+from burnman import constants
+from burnman import geotherm
+import warnings
+
+
+from .material import Material, material_property
+from .mineral import Mineral
+from .composite import Composite
+from .seismic import Seismic1DModel
+
+import matplotlib.pyplot as plt
+
+
+class Layer(object):
+    """
+    A planetary layer class
+    """
+
+    def __init__(self, name=None, radius_planet=None,
```

Comment:
Hmmm, I don't think the user should be able to know the depth from the surface of a planet when they've just only declared a layer... radius is a much more natural choice.

For a layer, we only need to know the top_radius, bottom_radius, and pressure and gravity at either the top or the bottom or one of each (which could conceivably be user choice, though that might be messy). Mass and gravity computations don't require that we know anything about the planet away from the boundaries (Gauss' law).